forte/data/data_store.py

```diff
@@ -250,6 +251,46 @@ def _new_group(self, type_name: str, member_type: str) -> List:
         entry += len(self._type_attributes[type_name]) * [None]
         return entry
 
+    def _is_subclass(self, type_name: str, cls) -> bool:
+        r"""This function takes a fully qualified ``type_name`` class name,
+        ``cls`` class and returns whether ``type_name``  class is the``cls``
+        subclass or not.
+
+        Args:
+            type_name: A fully qualified name of an entry class
+            cls: An entry class.
+
+        Returns:
+            A boolean value whether ``type_name``  class is the``cls``
+            subclass or not.
+
+        """
+        if type_name not in self._type_attributes:
+            self._type_attributes[type_name] = {}
+        if "parent_class" not in self._type_attributes[type_name]:
+            self._type_attributes[type_name]["parent_class"] = set()
+        cls_qualified_name = get_full_module_name(cls)
+        type_name_parent_class = self._type_attributes[type_name][
+            "parent_class"
+        ]
+        if len(type_name_parent_class) > 0:
+            return bool(cls_qualified_name in type_name_parent_class)
+        else:
+            entry_class = get_class(type_name)
+            if issubclass(entry_class, cls):
+                type_name_parent_class.add(cls_qualified_name)
+                return True
+            else:
+                cls_subclasses = cls.__subclasses__()
+                for cls_subclass in cls_subclasses:
+                    if issubclass(entry_class, cls_subclass):
+                        cls_subclass_qualified_name = get_full_module_name(
+                            cls_subclass
+                        )
+                        type_name_parent_class.add(cls_qualified_name)
+                        type_name_parent_class.add(cls_subclass_qualified_name)
+                return bool(cls_qualified_name in type_name_parent_class)
```

Comment:
This `else` block is a bit confusing to me. What are we trying to achieve here? `issubclass` check can check for any direct/indirect subclassing, listing the `subclass` of `cls` won't add anything new.